File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/SimulatorShutdownRequest.java
package com.linkpoint.slproto.messages;

import com.google.common.base.Ascii;
import com.linkpoint.slproto.SLMessage;
import java.nio.ByteBuffer;

public class SimulatorShutdownRequest extends SLMessage {
    public SimulatorShutdownRequest() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 4;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleSimulatorShutdownRequest(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put(Ascii.CR);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/SimulatorShutdownRequest.kt
package com.linkpoint.slproto.messages

import com.google.common.base.Ascii
import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class SimulatorShutdownRequest : SLMessage() {
    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 4
    }

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleSimulatorShutdownRequest(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.putShort(-1)
        buffer.put(0.toByte())
        buffer.put(Ascii.CR)
    }

    override fun UnpackPayload(buffer: ByteBuffer) {}
}